wdl/qc_imputation.wdl

```diff
@@ -93,24 +112,39 @@ workflow qc_imputation {
         f=f, het_sd=batch_qc.het_sd_threshold[0], pi_hat_min_n=batch_qc.pi_hat_min_n_threshold[0], docker=docker
     }
 
+    # for chip qc, filter variants and samples in batches as per qc and merge batches
+    scatter (i in range(length(vcfs))) {
+        call filter_batch {
+            input: docker=docker, bed=vcf_to_bed.out_bed[i], joint_qc_exclude_variants=gather_snpstats_joint_qc.exclude_variants_joint,
+            batch_qc_exclude_variants=batch_qc.variants_exclude[i], exclude_samples=duplicates.allbatches_samples_exclude
+        }
+    }
+    scatter (chr in chrs) {
+        call merge_batches {
+            input: docker=docker, name=name+"_chr"+chr, chr=chr, beds=filter_batch.out_bed, bims=filter_batch.out_bim, fams=filter_batch.out_fam
+        }
+        call post_subset_sub.subset_samples as subset_samples_chip {
+            input: vcfs=[merge_batches.vcf], vcf_idxs=[merge_batches.vcf_idx], already_excluded_samples=duplicates.allbatches_samples_exclude,
+            exclude_denials=exclude_denials, duplicate_samples=duplicate_samples, sample_summaries=joint_plots.sample_summaries,
+            docker=docker
+        }
+    }
+
```

Comment:
could finally merge the per chromosome files to single chip vcf  and plink files in the end and could add those as outputs of the workflow. Also the final imputation results could be added as workflow outputs for clarity

Reply:
I'm not sure if escience did this already but adding